File: conformance/repo_lint/vectors/fixtures/perl/docstring_test.pl
#!/usr/bin/env perl
# Perl test fixtures for repo_lint docstring validation
#
# Purpose: Test missing docstrings, correct docstrings, pragma exemptions,
# and edge cases for Perl symbol discovery via PPI.

use strict;
use warnings;

# This subroutine has documentation (should pass)
# Purpose: Demonstrates properly documented subroutine
sub function_with_doc {
    return "ok";
}

# Missing docstring - should be detected
sub function_without_doc {
    return "missing";
}

# Purpose: Properly documented subroutine with parameters
# Args:
#   $arg1 - First argument
#   $arg2 - Second argument
sub multiline_function {
    my ($arg1, $arg2) = @_;
    return "$arg1 $arg2";
}

# noqa: FUNCTION
sub exempted_function {
    return "pragma exemption";
}

package TestPackage;

# Purpose: Package-scoped subroutine
sub package_function {
    return "package scope";
}

1;

File: conformance/repo-lint/vectors/fixtures/python/docstring_test.py
"""Python test fixtures for repo_lint docstring validation.

:Purpose:
    Test missing docstrings, correct docstrings, pragma exemptions,
    and edge cases for Python symbol discovery and validation.

:Environment Variables:
    None

:Examples:
    This file is used by test_vectors.py for conformance testing::

        python3 -m tools.repo_lint.tests.test_vectors

:Exit Codes:
    N/A - This is a test fixture file, not executable
"""

from __future__ import annotations


def function_with_doc():
    """This function has a docstring (should pass)."""
    return "ok"


def no_doc():
    return "missing docstring"


class MissingClassDoc:
    def method_with_doc(self):
        """Method has documentation."""
        pass  # pylint: disable=unnecessary-pass

    def method_without_doc(self):
        pass  # pylint: disable=unnecessary-pass


class ProperlyDocumented:
    """Class with proper documentation.

    :Purpose:
        Demonstrates a properly documented class with all required sections.
    """

    def __init__(self):
        """Initialize the class.

        :Purpose:
            Constructor documentation.
        """
        pass  # pylint: disable=unnecessary-pass

    def public_method(self):
        """Public method with documentation.

        :Purpose:
            Demonstrates proper method documentation.
        """
        pass  # pylint: disable=unnecessary-pass


def exempted_function():  # noqa: D103
    return "pragma exemption"


def multiline_signature(
    arg1,
    arg2,
    arg3="default",
):
    """Function with multiline signature.

    :Purpose:
        Test multiline parameter lists.
    """
    return arg1 + arg2 + arg3
